src/search/SearchMessagesScreen.js

```diff
@@ -109,6 +105,13 @@ class SearchMessagesScreen extends PureComponent<Props, State> {
     });
   };
 
+  // The real work to be done on a query is async.  This wrapper exists
+  // just to fire off `handleQueryChangeInner` without waiting for it.
+  // TODO do we even need this wrapper?
```

Comment:
As written, I believe, the answer is "no". It may be that React Native will quietly *wait* on `Promise`s returned by event handlers (if only to funnel rejections through its usual error-reporting paths), but it certainly isn't going to *block* on them.